docs/registry/README.md

```diff
@@ -0,0 +1,385 @@
+for a consumer to create a Trigger without resorting to some other OOB mechanism.
+
+
+## Proposed Ideas
+
+### EventType CRD
+
+We propose introducing a namespaced-EventType CRD. Here is an example of how a CR would look like:
+
+```yaml
+apiVersion: eventing.knative.dev/v1alpha1
+kind: EventType
+metadata:
+  name: repopush
+  namespace: default
+spec:
+  type: repo:push
+  source: my-user/my-repo
+  schema: /my-schema
+  broker: default
+```
+
+- The `name` of the EventType is advisory, non-authoritative. Given that Cloud Event types can 
+contain characters that may not comply with Kubernetes naming conventions, we will (slightly) 
+modify those names to make them K8s-compliant, whenever we need to generate them. 
+
+- `type` is authoritative. This refers to the Cloud Event type as it enters into the eventing mesh. 
+
+- `source`: an identifier of where we receive the event from. This might not necessarily be the Cloud Event source 
+attribute. 
+
+If we have control over the entity emitting the Cloud Event, as is the case of many of our receive adaptors, 
+then we propose to add a Cloud Event custom extension (e.g., from) with this information, to ease the creation of filters 
+on Triggers later on.
+As the Cloud Event source attribute is somewhat useless (e.g., github pull requests are populated with `https://github.com/<owner>/<repo>/pull/<pull_id>`), 
+there is no way of doing exact matching of Cloud Event sources on Triggers. Thus, we propose adding this custom extension 
+to Cloud Events whenever we can. If the extension is not present, then we fallback to the Cloud Event source. 
+Note that when we start supporting more advanced filtering mechanisms on Triggers, we might not need this.
+
+
+- `schema` is a URI with the EventType schema. It may be a JSON schema, a protobuf schema, etc. It is optional.
+
+- `broker` refers to the Broker that can provide the EventType. 
+
+### Typical Flow
+
+`1.` A `Cluster Configurator` configures the cluster in a way that allows the population of EventTypes in the Registry. 
+We foresee the following three ways of populating the Registry so far:
+
+`1.1` Event Source CR installation
+
+Upon installation of an Event Source CR by a `Cluster Configurator`, the Source will register its EventTypes.
+
+Example:
+
+```yaml
+apiVersion: sources.eventing.knative.dev/v1alpha1
+kind: GitHubSource
+metadata:
+  name: github-source-sample
+  namespace: default
+spec:
+  eventTypes:
+    - push
+    - pull_request
+  ownerAndRepository: my-other-user/my-other-repo
+  accessToken:
+    secretKeyRef:
+      name: github-secret
+      key: accessToken
+  secretToken:
+    secretKeyRef:
+      name: github-secret
+      key: secretToken
+  sink:
+    apiVersion: eventing.knative.dev/v1alpha1
+    kind: Broker
+    name: default
+
+```
+ 
+By applying the above file, two EventTypes will be registered, with types `dev.knative.source.github.push` and 
```

Comment:
What code did this? If we expect the EventSource's controller to do it then let's say that.

Reply:
yes, will add that. Thanks!